examples/interactionsDemo.html

```diff
@@ -0,0 +1,532 @@
+                                                        case 'email':
+                                                            cellMediaType.classList += " fa fa-envelope";
+                                                            cellMediaType.setAttribute('aria-label', 'email');
+                                                            break;
+                                                        case 'voice':
+                                                        case 'callback':
+                                                            cellMediaType.classList += " fas fa-phone";
+                                                            cellMediaType.setAttribute('aria-label', 'voice or callback');
+                                                            break;
+                                                        case 'chat':
+                                                            cellMediaType.classList += " fas fa-comments";
+                                                            cellMediaType.setAttribute('aria-label', 'chat');
+                                                            break;
+                                                        case 'screenshare':
+                                                        case 'cobrowse':
+                                                            cellMediaType.classList += " fas fa-desktop";
+                                                            cellMediaType.setAttribute('aria-label', 'screenshare or cobrowse');
+                                                            break;
+                                                        default:
+                                                            if (typeof mediaType != 'undefined') {
+                                                                cellMediaType.innerHTML = mediaType;
+                                                                cellMediaType.paddingLeft = "0px";
+                                                            }
+                                                    }
+                                                    return true;
+                                                }
+                                            });
+
+                                            if (canViewWrapUpCodes) {
+                                                conversation.participants.forEach(function (participant) {
+                                                    participant.sessions.forEach(function (session) {
+                                                        session.segments.some(function (segment) {
+                                                            if (segment.hasOwnProperty('wrapUpCode') && typeof segment.wrapUpCode != 'undefined' && segment.wrapUpCode != '') {
+                                                                new platformClient.RoutingApi().getRoutingWrapupcode(segment.wrapUpCode)
```

Comment:
This nested promise worries me a bit from a memory leak perspective.   If there's a bunch of reloads with wrap-up codes, will those references start to stack up and fail to remove the row listeners; leading to a leak?

Now, this is a demo app so it would take a lot of atypical usage to start to accumulate memory and won't be used in production at that.  Also, the fix would probably be to decouple data acquisition from markup building (and cache routing code results) so that would probably balloon this effort out more than we need it to.

Reply:
Perhaps? When I tear down the table body on line 241 does that not free up that memory and remove the row listeners? I took some memory snapshots in chrome and didn't see any nodes without direct references to JS and detached from the DOM tree. It seemed to me that the table was getting garbage collected on reload. Although, I am not very experienced in using the memory dev tools.

If you are concerned I could pull out wrap up codes all together or do another refactor.

Reply:
Yeah, I was worried about both the click listener and the promise in terms of leaking.  I did a little research and it doesn't seem like listeners would hold a reference implicitly.  And, I wasn't sure if the memory would be free'd once the promise resolved if the dom was torn down before.  That was my major concern.

If it doesn't appear to be leaking, I'm fine with that as it is an example/testing tool after all.

Reply:
![image](https://user-images.githubusercontent.com/11987207/52669552-4d8cb900-2ee4-11e9-926d-a9449fe38385.png)

Reply:
Yeah, memory seems to be released on GC.  And listeners are growing but then shrinking back down and stable at 61 after quite a few runs.  With only 16 rows, that seems a bit high for a listener count, but it could be frameworks, etc.  Also, since it's not growing, that's good.

I wasn't gentle.  Clicked multiple times while the table was loading, so I thik we're ok from a leak standpoint.